dbms/src/Storages/MergeTree/BackgroundProcessingPool.cpp

```diff
@@ -36,7 +37,7 @@ void BackgroundProcessingPool::TaskInfo::wake()
         std::unique_lock<std::mutex> lock(pool.tasks_mutex);
 
         auto next_time_to_execute = iterator->first;
-        TaskHandle this_task_handle = iterator->second;
+        auto this_task_handle = iterator->second;
```

Comment:
This change doesn't make anything better.

Reply:
From appearance point of view it is better than `std::shared_ptr<BackgroundProcessingPoolTaskInfo> this_task_handle = iterator->second;`. (`TaskHandle` is unaccessable here after extracting `BackgroundProcessingPoolTaskInfo` from `BackgroundProcessingPool::Task`).

But it makes sense to use const references here.